### PR TITLE
DMs

### PR DESCRIPTION
**Pull request by emilianya**

### CODE REVIEW

**Comment by emilianya** [comment]
image
cat for reviewer

File: src/routes/v1/gateway.ts [hunk 1]
```diff
@@ -117,6 +118,16 @@ async function handleMessage(message, ws, user, socketId) {
...
                     let quark = await Quarks.findOne({_id: event.split("_")[1], members: user._id});
+                    if (ConstantID_DMvQuark.equals(event.split("_")[1]) || event.split("_")[1] === "dm") {
+                        quark = {
+                            name: "Direct Messages",
+                            _id: ConstantID_DMvQuark,
+                            members: [user._id],
+                            invite: "direct messages", // Impossible invite
+                            owners: [ ConstantID_SystemUser ],
+                            channels: []
+                        }
```
Comment (nyakase) [inline src/routes/v1/gateway.ts:129]: You paste this one billion trillion times, perhaps it should be a constant stored somewhere

**Review by nyakase — COMMENTED** [review]
I'm useless sorry

File: src/routes/v1/gateway.ts (hunk 1 above)
Comment (emilianya) [reply src/routes/v1/gateway.ts:129]: yeah ive been thinking about this

Comment (nyakase) [reply src/routes/v1/gateway.ts:129]: never had an original thought in my life

Comment (emilianya) [reply src/routes/v1/gateway.ts:129]: then i decided im lazy and didnt make it a constant because the user id and channels keep changing

**Review by nyakase — APPROVED** [review]
i dont understand any of this but you wrote it so it must be good NOO NOOO FIREFOX NOOOO STOP NOOO FIREFOX

**Comment by emilianya** [comment]
image
extra cat for accepted review

**Comment by nyakase** [comment]
me when i waste my money:

**Pull request merged**